src/Games/NexusMods.Games.RedEngine/Cyberpunk2077/SortOrder/RedMod/RedModSortableItemProviderFactory.cs

```diff
@@ -0,0 +1,78 @@
+using System.Diagnostics;
+using DynamicData;
+using NexusMods.Abstractions.Games;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.MnemonicDB.Abstractions;
+using R3;
+
+namespace NexusMods.Games.RedEngine.Cyberpunk2077.SortOrder;
+
+public class RedModSortableItemProviderFactory : ISortableItemProviderFactory
```

Comment:
It's not necessarily blocking for this PR, but I think most of this can be lifted out into an abstract class. `ARedModSortableItemProviderFactory`, if we're going by Halgari conventions of the codebase.

There doesn't seem to be anything specifically REDmod specific here; and it seems likely that this boilerplate will be repeated again. `RedModSortableItemProvider` is just an `ILoadoutSortableItemProvider`; so you could do `ARedModSortableItemProviderFactory<TLoadoutSortableItemProvider> where TLoadoutSortableItemProvider : ILoadoutSortableItemProvider` and have almost all the logic, short of `SortOrderName` and `StaticSortOrderTypeId` be moved to the abstract type.

Reply:
Yeah, it would just be `ASortableItemProviderFactory` without the redMod part, but that is the idea. 
I want to add the abstractions when I add support for the next load orders (`.archive` LO, BG3 pak LO etc), there will probably be more stuff to abstract away at that point.

The factory in particular is likely going to have to include a bunch of UI related info as well.